astropy/visualization/wcsaxes/ticklabels.py

```diff
@@ -39,17 +39,54 @@ def __init__(self, frame, *args, **kwargs):
 
     def clear(self):
         self.world = defaultdict(list)
-        self.pixel = defaultdict(list)
+        self.data = defaultdict(list)
         self.angle = defaultdict(list)
         self.text = defaultdict(list)
         self.disp = defaultdict(list)
 
-    def add(self, axis, world, pixel, angle, text, axis_displacement):
+    def add(self, axis=None, world=None, pixel=None, angle=None, text=None,
+            axis_displacement=None, data=None):
+        """
+        Add a label.
+
+        Parameters
+        ----------
+        axis : str
+            Axis to add label to.
+        world : Quantity
+            Coordinate value along this axis.
+        pixel : [float, float]
+            Pixel coordinates of the label. Deprecated and no longer used.
+        angle : float
+            Angle of the label.
+        text : str
+            Label text.
+        axis_displacement : float
+            Displacement from axis.
+        data : [float, float]
+            Data coordinates of the label.
+        """
+        required_args = ['axis', 'world', 'angle', 'text', 'axis_displacement', 'data']
+        if pixel is not None:
+            warnings.warn('Setting the pixel coordinates of a label does nohing and is deprecated, '
+                          'as these can only be accurately calculated when Matplotlib '
+                          'is drawing a figure. To prevent this warning pass the following '
+                          f'arguments as keyword arguments: {required_args}',
+                          AstropyDeprecationWarning)
+        if (axis is None or
+            world is None or
+            angle is None or
+            text is None or
+            axis_displacement is None or
```

Comment:
Isn't it supposed to be like this or is this now `black` does things?
```suggestion
                world is None or
                angle is None or
                text is None or
                axis_displacement is None or
```

Reply:
`black` would format this as:
```python
if (
    axis is None
    or world is None
    or angle is None
    or text is None
    or axis_displacement is None
    or data is None
):
```